Fix overlapping src and dst for strlcpy that caused crash on launch.
https://sourceforge.net/p/xine/xine-ui/ci/922dfad300c48592c23bffe647035a2a2b1c82b3/
--- src/xitk/xine-toolkit/backend_x11.c.orig
+++ src/xitk/xine-toolkit/backend_x11.c
@@ -2248,9 +2248,9 @@
       const char *res_class = (const char *)win->props[XITK_X11_WT_RES_CLASS].value;
       const char *res_name = (const char *)win->props[XITK_X11_WT_RES_NAME].value;
 
-      if (res_class)
+      if (res_class && res_class != win->res_class)
         strlcpy (win->res_class, res_class, sizeof (win->res_class));
-      if (res_name)
+      if (res_name && res_name != win->res_name)
         strlcpy (win->res_name, res_name, sizeof (win->res_name));
       xclasshint->res_name  = win->res_name[0] ? win->res_name : "Xine Window";
       xclasshint->res_class = win->res_class[0] ? win->res_class : (char *)"Xitk";
